code/modules/vehicles/spacepod/equipment/sensors.dm

```diff
@@ -0,0 +1,78 @@
+/obj/item/pod_equipment/sensors
+	slot = POD_SLOT_SENSORS
+	name = "sensor suite"
+	desc = "A sensor suite for space pods, containing some tech doodads and a built-in GPS. Manufacted by Nanotrasen, and has weird labels like \"Not for infowar\" or \"manufacted on Tau Ceti IV\"."
+	icon_state = "sensors"
+	/// reference to our GPS component
+	var/datum/component/gps/item/gps
+	/// traits given to all passengers
+	var/traits_given
+
+/obj/item/pod_equipment/sensors/on_attach(mob/user)
+	. = ..()
+	gps = AddComponent(/datum/component/gps/item, "POD[rand(0,999)]", state = GLOB.in_vehicle_state, overlay_state = FALSE)
+	gps.tracking = FALSE
+	START_PROCESSING(SSobj, src)
+	if(islist(traits_given))
+		RegisterSignal(pod, COMSIG_VEHICLE_OCCUPANT_ADDED, PROC_REF(occupant_added))
+		RegisterSignal(pod, COMSIG_VEHICLE_OCCUPANT_REMOVED, PROC_REF(occupant_removed))
+		for(var/occupant in pod.occupants)
+			occupant_added(occupant)
+
+/obj/item/pod_equipment/sensors/on_detach(mob/user)
+	. = ..()
+	QDEL_NULL(gps)
+	UnregisterSignal(pod, list(COMSIG_VEHICLE_OCCUPANT_ADDED, COMSIG_VEHICLE_OCCUPANT_REMOVED))
+	if(islist(traits_given))
+		for(var/occupant in pod.occupants)
+			occupant_removed(occupant)
+
+/obj/item/pod_equipment/sensors/proc/occupant_added(datum/source, mob/living/carbon/occupant, flags)
+	SIGNAL_HANDLER
+	if(istype(occupant))
+		occupant.add_traits(traits_given, REF(src))
+		occupant.update_sight()
```

Comment:
why?

Reply:
this only runs if traits_given is a list
traits_given is either not a list (base type)
or contains a sight trait (TRAIT_MESON_VISION, TRAIT_TRUE_NIGHT_VISION) which we add and then run update_sight() so it actually does anything